view.php

```diff
@@ -52,31 +52,43 @@
 
     echo $OUTPUT->header();
 
-    $entries = $DB->get_records('cpdlogbook_entries', ['cpdlogbookid' => $cm->instance]);
-    $sum = 0;
-    foreach ($entries as $e) {
-        $sum += $e->points;
+    $periodid = period::get_period_for_date(time(), $cm->instance);
+
+    if ($periodid != 0) {
+        $period = new period($periodid);
+
+        $entries = $DB->get_records('cpdlogbook_entries', ['cpdlogbookid' => $cm->instance, 'periodid' => $periodid]);
```

Comment:
You should be able to do the sum in sql. You could probably also refactor this into a helper somewhere, possibly in the period persitent